From 8dab90fc1b32b1a28f9938ed1323f1db66da4116 Mon Sep 17 00:00:00 2001
From: Liang Chen <cl@rock-chips.com>
Date: Sun, 25 Feb 2018 14:25:20 +0800
Subject: [PATCH] soc: rockchip: opp_select: make scale sel do not depend on
 leakage

When get leakage failed, we assume the leakage is zero.

Change-Id: I7731eaaa6dc31620d0210c1c9138631b8890be8d
Signed-off-by: Liang Chen <cl@rock-chips.com>
---
 drivers/soc/rockchip/rockchip_opp_select.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/soc/rockchip/rockchip_opp_select.c b/drivers/soc/rockchip/rockchip_opp_select.c
index e4f2ff38da75..22b07531d2d4 100644
--- a/drivers/soc/rockchip/rockchip_opp_select.c
+++ b/drivers/soc/rockchip/rockchip_opp_select.c
@@ -270,6 +270,14 @@ int rockchip_of_get_lkg_scale_sel(struct device *dev, char *name)
 			dev_info(dev, "%s-scale-sel=%d\n", name, volt_sel);
 			return volt_sel;
 		}
+	} else {
+		dev_info(dev, "get %s fail\n", name);
+		ret = rockchip_get_volt_sel(np, "rockchip,leakage-scaling-sel",
+					    0, &volt_sel);
+		if (!ret) {
+			dev_info(dev, "%s-scale-sel=%d\n", name, volt_sel);
+			return volt_sel;
+		}
 	}
 
 	return ret;
-- 
2.35.3

